src/events/messageCreate/checkAvailability.js

```diff
@@ -0,0 +1,252 @@
+            },
+        },
+    },
+    {
+        'type':     'function',
+        'function': {
+            'name':        'unableToParse',
+            'description': 'If the user inputs content that cannot be used in the other functions.',
+            'parameters':  {},
+        },
+    },
+];
+
+
+const examples =
+   `{
+        'role':    'user',
+        'content': I am busy from 10-12 tommorow. Message date 6/12/2024.,
+    },
+    {
+        'role':    'assistant',
+        'content': {name: 'rememberUnavailability', arguments: '{"times":[{"start":"2024-06-12T10:00:00","end":"2024-06-12T14:00:00","reason":"busy"}]}"}',
+    },
+    {
+        'role':    'user',
+        'content': I am busy all day this upcoming wednesday, thursday i am busy from 10-2 and friday 10-4. Message date 6/11/2024.,
+    },
+    {
+        'role':    'assistant',
+        'content': {name: 'rememberUnavailability', arguments: '{"times":[{"start":"2024-06-12T00:00:00","end":"2024-06-12T23:59:59","reason":"busy"}, {"start":"2024-06-13T10:00:00","end":"2024-06-13T14:00:00","reason":"busy"}, {"start":"2024-06-14T10:00:00","end":"2024-06-14T16:00:00","reason":"busy"}]}"}',
+    },
+    {
+        'role':    'user',
+        'content': I free from 9-5 monday through friday. Message date 6/12/2024,
+    },
+    {
+        'role':    'assistant',
+        'content': {name: "rememberAvailability", arguments: "{"from":"09:00","to":"17:00","days":[true,true,true,true,true]}",},
+    },
+    `;
+
+/**
+ * Given a function object from open ai and a message, attempt to parse 
+ * @param {DiscordJS Message} message 
+ * @param {OpenAI Tool} calledFunction 
+ */
+const parseResults = (message, calledFunction) =>{
+    const action = {
+        'rememberUnavailability': () => rememberUnavailability(message, JSON.parse(calledFunction.arguments)),
+        'rememberAvailability':   () => rememberAvailability(message, JSON.parse(calledFunction.arguments)),
+        'unableToParse':          () => unableToParse(message),
+    };
+    action[calledFunction.name]();
+};
+
+// checks the message sent in the channel and sends it to openai to parse, then if possible saves the availability data
+module.exports = async (client, message) => {
+    try {
+        const availabilityChannel = await getAvailabilityChannel();
+
+        // only send a message if it's not from a bot and it's from the available channel
+        if (message.author.bot || !availabilityChannel || message.channelId !== availabilityChannel.id) {
+            return;
+        }
+        const date = new Date(message.createdTimestamp).toLocaleDateString();
+        const prompt = stripIndents`Based on the attached message and examples, 
+        use one of the provided tools to parse unavailability using 'rememberUnavailability' or availability using 'rememberAvailability'. 
+        If either don't work call 'unableToParse' from the tools.The message date is ${date}.`;
```

Comment:
Yeah, looking at it, I do suggest you define what availability and unavailability mean in precise terms.